roles/grafana_agent/tasks/main.yml

```diff
@@ -0,0 +1,51 @@
+---
+- name: Include secrets
+  include_vars: "{{ secrets_path | mandatory }}/mimir_password.yml"
+  no_log: true
+  tags:
+    - always
+
+- name: "Import Grafana GPG key"
+  become: true
+  ansible.builtin.apt_key:
+    url: "https://apt.grafana.com/gpg.key"
+    state: "present"
+  when: ansible_pkg_mgr == "apt"
+
+- name: "Add Grafana's repository to APT sources list"
+  become: true
+  ansible.builtin.apt_repository:
+    repo: "deb https://apt.grafana.com stable main"
+    state: present
+  when: ansible_pkg_mgr == "apt"
+
+- name: "Add Grafana's repository to yum/dnf systems"
+  become: true
+  ansible.builtin.template:
+    src: "templates/grafana_repo.yml.j2"
+    dest: "/etc/yum.repos.d/grafana.repo"
+    mode: "0655"
+  when: ansible_os_family == "RedHat"
+
+- name: "Install grafana-agent"
+  become: true
+  ansible.builtin.package:
+    name: "grafana-agent"
+    state: "present"
+
+- name: "Enable grafana-agent"
+  become: true
+  ansible.builtin.service:
+    name: "grafana-agent"
+    state: "started"
+    enabled: true
+
+# Deploy config file from template and restart the agent      
+- name: "Configure agent"
+  become: true
+  ansible.builtin.template:
+    src: "templates/linux_agent_config.yml.j2"
+    dest: "/etc/grafana-agent.yaml"
+    mode: "0550"
```

Comment:
I recommend renaming the template to `grafana-agent.yaml.j2` so that it matches what you write to /etc. When those file names do not match, it makes it more confusing to maintain this codebase.

We should have a `user` parameter here.

`mode: "0550"` does not look right here. This file should not be executable. Probably `mode: "0440"` would be right, assuming `user` will be `root`.